project/src/components/private-route/private-route.tsx

```diff
@@ -0,0 +1,26 @@
+import {Route, Redirect} from 'react-router-dom';
+import {RouteProps} from 'react-router-dom';
+import {AppRoute, AuthorizationStatus} from '../../const';
+
+type PrivateRouteProps = RouteProps & {
+  render: () => JSX.Element;
+  authorizationStatus: AuthorizationStatus;
+}
+
+function PrivateRoute(props: PrivateRouteProps): JSX.Element {
+  const {exact, path, render, authorizationStatus} = props;
+
+  return (
+    <Route
+      exact={exact}
+      path={path}
+      render={() => (
+        authorizationStatus === AuthorizationStatus.Auth
+          ? render()
+          : <Redirect to={AppRoute.Login} />
+      )}
+    />
```

Comment:
А зачем нам этот `render`, если мы там и так берём все пропсы Роута? Т.е. мы можем делать то же, что и с Роутом

```suggestion
  authorizationStatus: AuthorizationStatus;
}

function PrivateRoute(props: PrivateRouteProps): JSX.Element {
  const {exact, path, authorizationStatus, component, childen,} = props;

  return (
    <Route exact={exact} path={path} component={component}>
      {authorizationStatus === AuthorizationStatus.Auth ? (
        children
      ) : (
        <Redirect to={AppRoute.Login} />
      )}
    </Route>
```

Соответственно в `app` подключать либо вместо
```tsx
          // render={() => <WinScreen />}
          component={WinScreen}
``` 
либо
```tsx
        <PrivateRoute
          exact
          path={AppRoute.Result}
          authorizationStatus={AuthorizationStatus.NoAuth}
        >
          <WinScreen />
        </PrivateRoute>
```